From 9cb0f0a68c29ff8905a9fc6906694b8b91a92695 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 19 Jul 2011 10:22:45 -0500
Subject: [PATCH 1/4] mx51evk enable ext2 support

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 include/configs/mx51evk.h |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/include/configs/mx51evk.h b/include/configs/mx51evk.h
index 3b18a18..8b9629e 100644
--- a/include/configs/mx51evk.h
+++ b/include/configs/mx51evk.h
@@ -91,6 +91,7 @@
 
 #define CONFIG_CMD_MMC
 #define CONFIG_GENERIC_MMC
+#define CONFIG_CMD_EXT2		/* EXT2 Support			*/
 #define CONFIG_CMD_FAT
 #define CONFIG_DOS_PARTITION
 
-- 
1.7.9.1

