From 98efa446666e381ff194a413ef2bce5450761aa7 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Fri, 23 Mar 2012 16:33:56 -0500
Subject: [PATCH] beagle make ulcd configure correctly out of the box

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 board/ti/beagle/beagle.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/board/ti/beagle/beagle.c b/board/ti/beagle/beagle.c
index 9bf884d..06873a3 100644
--- a/board/ti/beagle/beagle.c
+++ b/board/ti/beagle/beagle.c
@@ -405,8 +405,10 @@ int misc_init_r(void)
 	case BBTOYS_ULCD:
 		printf("Recognized BeagleBoardToys uLCD-lite board\n");
 		setenv("buddy2", "bbtoys-ulcd");
-		setenv("defaultdisplay", "lcd");
-		setenv("uenvcmd", "i2c dev 1 ; i2c mw 40 00 00; i2c mw 40 04 80; i2c mw 40 0d 05; i2c mw 40 0d 15");
+		setenv("defaultdisplay", "dvi");
+		setenv("dvimode", "800x480MR-16@60");
+		setenv("lcd1", "i2c mw 40 00 00; i2c mw 40 04 80; i2c mw 40 0d 05");
+		setenv("uenvcmd", "i2c dev 1 ; run lcd1; i2c dev 0");
 		break;
 	case BEAGLE_NO_EEPROM:
 		printf("No EEPROM on expansion board\n");
-- 
1.7.9.1

