From 90c12f36b3cf8e654437c8e10031de3e1246c156 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Wed, 22 Dec 2010 16:27:14 -0600
Subject: [PATCH] arm omap beagle older bx boards need more time

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/mmc/omap_hsmmc.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/omap_hsmmc.c b/drivers/mmc/omap_hsmmc.c
index 6f2280a..31e758c 100644
--- a/drivers/mmc/omap_hsmmc.c
+++ b/drivers/mmc/omap_hsmmc.c
@@ -31,8 +31,8 @@
 #include <asm/io.h>
 #include <asm/arch/mmc_host_def.h>
 
-/* If we fail after 1 second wait, something is really bad */
-#define MAX_RETRY_MS	1000
+/* If we fail after 1.5 second wait, something is really bad */
+#define MAX_RETRY_MS	1500
 
 static int mmc_read_data(hsmmc_t *mmc_base, char *buf, unsigned int size);
 static int mmc_write_data(hsmmc_t *mmc_base, const char *buf, unsigned int siz);
-- 
1.7.1

