From 6461b5904e9e83f3ffa9098817e35064a8a36a6e Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Fri, 27 Jan 2012 08:31:24 -0600
Subject: [PATCH 1/2] beagle c4 should not be showing up with 1GB ram

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 board/ti/beagle/beagle.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/board/ti/beagle/beagle.c b/board/ti/beagle/beagle.c
index 6a457cb..9aa05d4 100644
--- a/board/ti/beagle/beagle.c
+++ b/board/ti/beagle/beagle.c
@@ -159,8 +159,8 @@ void get_board_mem_timings(u32 *mcfg, u32 *ctrla, u32 *ctrlb, u32 *rfr_ctrl,
 	switch (get_board_revision()) {
 	case REVISION_C4:
 		if (pop_mfr == NAND_MFR_STMICRO && pop_id == 0xba) {
-			/* 512MB DDR */
-			*mcfg = NUMONYX_V_MCFG_165(512 << 20);
+			/* Beagleboard Rev C4, 256MB DDR */
+			*mcfg = NUMONYX_V_MCFG_165(256 << 20);
 			*ctrla = NUMONYX_V_ACTIMA_165;
 			*ctrlb = NUMONYX_V_ACTIMB_165;
 			*rfr_ctrl = SDP_3430_SDRC_RFR_CTRL_165MHz;
-- 
1.7.4.1

