From a94dc9b827588df9b336942c30da1b938d3a6efb Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 19 Jul 2011 10:23:28 -0500
Subject: [PATCH 2/2] mx51evk use partition 1

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 include/configs/mx51evk.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/include/configs/mx51evk.h b/include/configs/mx51evk.h
index fe92984..1c73ebf 100644
--- a/include/configs/mx51evk.h
+++ b/include/configs/mx51evk.h
@@ -132,7 +132,7 @@
 	"script=boot.scr\0" \
 	"uimage=uImage\0" \
 	"mmcdev=0\0" \
-	"mmcpart=2\0" \
+	"mmcpart=1\0" \
 	"mmcroot=/dev/mmcblk0p3 rw\0" \
 	"mmcrootfstype=ext3 rootwait\0" \
 	"mmcargs=setenv bootargs console=ttymxc0,${baudrate} " \
-- 
1.7.4.1

