diff --git a/drivers/video/ipu_common.c b/drivers/video/ipu_common.c
index 36d4b23..db50484 100644
--- a/drivers/video/ipu_common.c
+++ b/drivers/video/ipu_common.c
@@ -352,7 +352,7 @@ static int ipu_pixel_clk_set_rate(struct clk *clk, unsigned long rate)
 	 */
 	__raw_writel((div / 16) << 16, DI_BS_CLKGEN1(clk->id));
 
-	clk->rate = (u64)(clk->parent->rate * 16) / div;
+	clk->rate = lldiv((u64)(clk->parent->rate * 16), div);
 
 	return 0;
 }
