From c5f8e8a18ce5574392512c514cbc6b9da8aa6634 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Wed, 6 Feb 2013 11:11:12 -0600
Subject: [PATCH 3/3] imx: mx6qsabre_common uEnv.txt

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 include/configs/mx6qsabre_common.h |    9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/include/configs/mx6qsabre_common.h b/include/configs/mx6qsabre_common.h
index 25f555f..63668e9 100644
--- a/include/configs/mx6qsabre_common.h
+++ b/include/configs/mx6qsabre_common.h
@@ -87,6 +87,7 @@
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	"script=boot.scr\0" \
+	"bootenv=uEnv.txt\0" \
 	"uimage=uImage\0" \
 	"fdt_file=" CONFIG_DEFAULT_FDT_FILE "\0" \
 	"fdt_addr=0x11000000\0" \
@@ -101,7 +102,9 @@
 	"mmcargs=setenv bootargs console=${console},${baudrate} " \
 		"root=${mmcroot}\0" \
 	"loadbootscript=" \
-		"fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${script};\0" \
+		"load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${bootenv};\0" \
+	"importbootenv=echo Importing environment from mmc ...; " \
+		"env import -t ${loadaddr} ${filesize}\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
 		"source\0" \
 	"loaduimage=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${uimage}\0" \
@@ -150,7 +153,9 @@
 	"mmc dev ${mmcdev};" \
 	"if mmc rescan ${mmcdev}; then " \
 		"if run loadbootscript; then " \
-		"run bootscript; " \
+			"echo Loaded environment from ${bootenv}; " \
+			"run importbootenv; " \
+			"run loaduimage; " \
 		"else " \
 			"if run loaduimage; then " \
 				"run mmcboot; " \
-- 
1.7.10.4

