From e2e24f7013ff9bbdee1751ae5988beb4e5f08f6b Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Fri, 15 Mar 2013 10:17:21 -0500
Subject: [PATCH] rpi_b: uEnv.txt, bootz, n fixes

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 include/configs/rpi_b.h |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/include/configs/rpi_b.h b/include/configs/rpi_b.h
index cf62e45..c1c01fb 100644
--- a/include/configs/rpi_b.h
+++ b/include/configs/rpi_b.h
@@ -86,7 +86,13 @@
 
 /* Commands */
 #include <config_cmd_default.h>
-#define CONFIG_CMD_BOOTZ
+#define CONFIG_FS_FAT
+#define CONFIG_FS_EXT4
+#define CONFIG_CMD_FS_GENERIC			/* Generic load commands */
+#define CONFIG_CMD_BOOTZ			/* bootz zImage support */
+#define CONFIG_SUPPORT_RAW_INITRD		/* bootz raw initrd support */
+#define CONFIG_CMD_FAT
+#define CONFIG_CMD_EXT2
 #define CONFIG_CMD_GPIO
 /* Some things don't make sense on this HW or yet */
 #undef CONFIG_CMD_FPGA
-- 
1.7.10.4

