From 710019759bce6bad04553591e70c8755cc17b7b8 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Fri, 15 Mar 2013 10:16:03 -0500
Subject: [PATCH] mx51evk: uEnv.txt, bootz, n fixes

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 include/configs/mx51evk.h |   27 ++++++++++++++++++---------
 1 file changed, 18 insertions(+), 9 deletions(-)

diff --git a/include/configs/mx51evk.h b/include/configs/mx51evk.h
index cb3d938..30b84b0 100644
--- a/include/configs/mx51evk.h
+++ b/include/configs/mx51evk.h
@@ -85,8 +85,14 @@
 #define CONFIG_MMC
 
 #define CONFIG_CMD_MMC
+#define CONFIG_FS_FAT
+#define CONFIG_FS_EXT4
+#define CONFIG_CMD_FS_GENERIC			/* Generic load commands */
+#define CONFIG_CMD_BOOTZ			/* bootz zImage support */
+#define CONFIG_SUPPORT_RAW_INITRD		/* bootz raw initrd support */
 #define CONFIG_GENERIC_MMC
 #define CONFIG_CMD_FAT
+#define CONFIG_CMD_EXT2
 #define CONFIG_DOS_PARTITION
 
 /*
@@ -152,20 +158,21 @@
 #define CONFIG_LOADADDR		0x90800000	/* loadaddr env var */
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
-	"script=boot.scr\0" \
+	"dtb_file=imx51-babbage.dtb\0" \
+	"bootenv=uEnv.txt\0" \
 	"uimage=uImage\0" \
 	"mmcdev=0\0" \
-	"mmcpart=2\0" \
-	"mmcroot=/dev/mmcblk0p3 rw\0" \
-	"mmcrootfstype=ext3 rootwait\0" \
+	"mmcpart=1\0" \
+	"mmcroot=/dev/mmcblk0p2 ro\0" \
+	"mmcrootfstype=ext4 rootwait\0" \
 	"mmcargs=setenv bootargs console=ttymxc0,${baudrate} " \
 		"root=${mmcroot} " \
 		"rootfstype=${mmcrootfstype}\0" \
 	"loadbootscript=" \
-		"fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${script};\0" \
-	"bootscript=echo Running bootscript from mmc ...; " \
-		"source\0" \
-	"loaduimage=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${uimage}\0" \
+		"load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${bootenv};\0" \
+	"importbootenv=echo Importing environment from mmc ...; " \
+		"env import -t ${loadaddr} ${filesize}\0" \
+	"loaduimage=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${uimage}\0" \
 	"mmcboot=echo Booting from mmc ...; " \
 		"run mmcargs; " \
 		"bootm\0" \
@@ -179,7 +186,9 @@
 #define CONFIG_BOOTCOMMAND \
 	"mmc dev ${mmcdev}; if mmc rescan; then " \
 		"if run loadbootscript; then " \
-			"run bootscript; " \
+			"echo Loaded environment from ${bootenv}; " \
+			"run importbootenv; " \
+			"run loaduimage; " \
 		"else " \
 			"if run loaduimage; then " \
 				"run mmcboot; " \
-- 
1.7.10.4

