From 67fbe5d74a152b2ef61b48c7352fd55b15b6d008 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 28 Aug 2012 10:05:34 -0500
Subject: [PATCH 4/4] panda: set dtb_file based on core

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 board/ti/panda/panda.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/board/ti/panda/panda.c b/board/ti/panda/panda.c
index ca4b8b3..e853033 100644
--- a/board/ti/panda/panda.c
+++ b/board/ti/panda/panda.c
@@ -128,6 +128,12 @@ int misc_init_r(void)
 
 	writel(altclksrc, &scrm->altclksrc);
 
+	if (omap_revision() < OMAP4460_ES1_0) {
+		setenv("dtb_file", "omap4-panda.dtb");
+	} else {
+		setenv("dtb_file", "omap4-pandaES.dtb");
+	}
+
 	return 0;
 }
 
-- 
1.7.10.4

