From 2ef78ff5bffe600583d8bcb309cf7f61973fc2bf Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 8 Nov 2012 19:06:10 -0600
Subject: [PATCH 3/3] panda: temp: enable pads and clocks for kernel

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 include/configs/omap4_common.h |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/configs/omap4_common.h b/include/configs/omap4_common.h
index cbc9bdb..c7c55f5 100644
--- a/include/configs/omap4_common.h
+++ b/include/configs/omap4_common.h
@@ -271,4 +271,7 @@
 
 #define CONFIG_SYS_THUMB_BUILD
 
+#define CONFIG_SYS_ENABLE_PADS_ALL
+#define CONFIG_SYS_CLOCKS_ENABLE_ALL
+
 #endif /* __CONFIG_OMAP4_COMMON_H */
-- 
1.7.10.4

