From c8de053c1c1c599c8b67a8e97e983dcb6091efec Mon Sep 17 00:00:00 2001
From: Manel Caro <mcaro@iseebcn.com>
Date: Tue, 7 Jan 2014 18:04:45 +0100
Subject: [PATCH 03/17] IGEPv5: Add GPIO Expander tca641x support

---
 drivers/gpio/tca642x.c | 20 +++++++++++++++++---
 1 file changed, 17 insertions(+), 3 deletions(-)

diff --git a/drivers/gpio/tca642x.c b/drivers/gpio/tca642x.c
index 6386835..de378d7 100644
--- a/drivers/gpio/tca642x.c
+++ b/drivers/gpio/tca642x.c
@@ -24,6 +24,8 @@
 #include <i2c.h>
 #include <tca642x.h>
 
+#ifndef CONFIG_TCA641X
+#define MAX_TCA64x_REG	3
 /* tca642x register address definitions */
 struct tca642x_bank_info tca642x_regs[] = {
 	{ .input_reg = 0x00,
@@ -39,6 +41,19 @@ struct tca642x_bank_info tca642x_regs[] = {
 	  .polarity_reg = 0x0a,
 	  .configuration_reg = 0x0e },
 };
+#else
+#define MAX_TCA64x_REG	2
+struct tca642x_bank_info tca642x_regs[] = {
+	{ .input_reg = 0x00,
+	  .output_reg = 0x02,
+          .polarity_reg = 0x04,
+          .configuration_reg = 0x06 },
+        { .input_reg = 0x01,
+          .output_reg = 0x03,
+          .polarity_reg = 0x05,
+          .configuration_reg = 0x07 },
+};
+#endif
 
 /*
  * Modify masked bits in register
@@ -148,7 +163,7 @@ int tca642x_set_inital_state(uchar chip, struct tca642x_bank_info init_data[])
 	uint8_t polarity_reg;
 	uint8_t output_reg;
 
-	for (i = 0; i < 3; i++) {
+	for (i = 0; i < MAX_TCA64x_REG; i++) {
 		config_reg = tca642x_regs[i].configuration_reg;
 		ret = tca642x_reg_write(chip, config_reg, 0xff,
 				init_data[i].configuration_reg);
@@ -159,7 +174,6 @@ int tca642x_set_inital_state(uchar chip, struct tca642x_bank_info init_data[])
 		ret = tca642x_reg_write(chip, output_reg, 0xff,
 				init_data[i].output_reg);
 	}
-
 	return ret;
 }
 
@@ -173,7 +187,7 @@ static int tca642x_info(uchar chip)
 	uint8_t data;
 
 	printf("tca642x@ 0x%x (%d pins):\n", chip, 24);
-	for (i = 0; i < 3; i++) {
+	for (i = 0; i < MAX_TCA64x_REG ; i++) {
 		printf("Bank %i\n", i);
 		if (tca642x_reg_read(chip,
 				     tca642x_regs[i].configuration_reg,
-- 
2.7.0

