From 7572e83f9c2cc4b33401540d3aa3b82afdeb4342 Mon Sep 17 00:00:00 2001
From: Manel Caro <mcaro@iseebcn.com>
Date: Mon, 20 Jan 2014 13:36:10 +0100
Subject: [PATCH 12/17] IGEPv5: Memory setup Change the define name

---
 board/isee/igep0050/board_configuration.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/board/isee/igep0050/board_configuration.c b/board/isee/igep0050/board_configuration.c
index ff2b3e3..677db01 100644
--- a/board/isee/igep0050/board_configuration.c
+++ b/board/isee/igep0050/board_configuration.c
@@ -97,14 +97,14 @@ struct igepv5_eeprom_config igepv5_config = {
     .lisa_regs = {
         .dmm_lisa_map_0 = DMM_LISA_MAP0,
         .dmm_lisa_map_1 = DMM_LISA_MAP1,
-#ifdef __IGEPv5_MEMORY_4G
+#ifdef IGEPv5_MEMORY_4G
         .dmm_lisa_map_2 = DMM_LISA_MAP2,    /* DMM_LISA_MAP2 = 4 GiB (is_ma_hm_interleave=1), DMM_LISA_MAP2_1G = 1 GiB (is_ma_hm_interleave=0)*/
 #else
 	.dmm_lisa_map_2 = DMM_LISA_MAP2_1G,    /* DMM_LISA_MAP2 = 4 GiB (is_ma_hm_interleave=1), DMM_LISA_MAP2_1G = 1 GiB (is_ma_hm_interleave=0)*/
 #endif
         .dmm_lisa_map_3 = DMM_LISA_MAP3,
         .is_ma_present	= 0x1,
-#ifdef __IGEPv5_MEMORY_4G
+#ifdef IGEPv5_MEMORY_4G
         .is_ma_hm_interleave = 0x1,
 #else
 	.is_ma_hm_interleave = 0x0,
-- 
2.7.0

