From 068104be978bf88a3f29acb154113950d760e2d5 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Wed, 21 Jun 2017 09:25:09 -0500
Subject: [PATCH 2/2] NFM: board_is_x15_revb1_blank

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 board/ti/am57xx/board.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/board/ti/am57xx/board.c b/board/ti/am57xx/board.c
index ab27468e5c..b92facd346 100644
--- a/board/ti/am57xx/board.c
+++ b/board/ti/am57xx/board.c
@@ -48,9 +48,9 @@
 #define board_is_am571x_sndrblock()	board_ti_is("AM571SBA")
 
 //Blank Flasher
-//#define board_is_x15_revb1_blank()	1
+#define board_is_x15_revb1_blank()	1
 //Normal
-#define board_is_x15_revb1_blank()	board_ti_is("XM57BLNK")
+//#define board_is_x15_revb1_blank()	board_ti_is("XM57BLNK")
 
 //Blank Flasher
 //#define board_is_x15_revc_blank()	1
-- 
2.11.0

