From 5e4d30e23a3adc9841b824d938abc556f9bc2201 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Wed, 21 Jun 2017 09:26:36 -0500
Subject: [PATCH 2/2] NFM: board_is_am571x_blank

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 board/ti/am57xx/board.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/board/ti/am57xx/board.c b/board/ti/am57xx/board.c
index ab27468e5c..53143b5b38 100644
--- a/board/ti/am57xx/board.c
+++ b/board/ti/am57xx/board.c
@@ -58,9 +58,9 @@
 #define board_is_x15_revc_blank()	board_ti_is("XM57BLNK")
 
 //Blank Flasher
-//#define board_is_am571x_blank()		1
+#define board_is_am571x_blank()		1
 //Normal
-#define board_is_am571x_blank()		board_ti_is("XM57BLNK")
+//#define board_is_am571x_blank()		board_ti_is("XM57BLNK")
 
 #ifdef CONFIG_DRIVER_TI_CPSW
 #include <cpsw.h>
-- 
2.11.0

