From a69084be092573bd5ed5b2b221599abd1ebafecf Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 16 May 2017 10:20:50 -0500
Subject: [PATCH 2/2] NFM: board_is_x15_revb1_blank

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 board/ti/am57xx/board.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/board/ti/am57xx/board.c b/board/ti/am57xx/board.c
index 50d05ea747..16e83fe3de 100644
--- a/board/ti/am57xx/board.c
+++ b/board/ti/am57xx/board.c
@@ -46,9 +46,9 @@
 #define board_is_am571x_sndrblock()	board_ti_is("AM571SBA")
 
 //Blank Flasher
-//#define board_is_x15_revb1_blank()	1
+#define board_is_x15_revb1_blank()	1
 //Normal
-#define board_is_x15_revb1_blank()	board_ti_is("XM57BLNK")
+//#define board_is_x15_revb1_blank()	board_ti_is("XM57BLNK")
 
 //Blank Flasher
 //#define board_is_am571x_blank()		1
-- 
2.11.0

